src/typeDefinitions/role.types.d.ts

```diff
@@ -0,0 +1,20 @@
+export type GuildRole = {
+  id: string;
+  name: string;
+};
+
+export type GuildDetails = {
+  // This type defines the partial guild details response that we get from discord
+  id: string;
+  name: string;
+  roles: Array<{
+    id: string;
+    name: string;
+    permissions: string;
+    position: number;
+    color: number;
+    hoist: boolean;
+    managed: boolean;
+    mentionable: boolean;
+  }>;
```

Comment:
why do we have this angular bracket in end?

Reply:
```
 Array<{
    id: string;
    name: string;
    permissions: string;
    position: number;
    color: number;
    hoist: boolean;
    managed: boolean;
    mentionable: boolean;
  }>;
```

If you see the above code, in between angular brackets I have defined a type. 

Since Array here is a generic hence we can pass type between angular brackets to define the type of each element in Array.